### PR TITLE
v1.2.0

### PR DESCRIPTION
**Pull request by igcherkaev**
__IMPORTANT__: Configuration scheme changed to mark token and password as secrets, if you were using st2 datastore, you might need to encrypt the values!

* Add new ``github.get_contents`` action which allows user to retrieve file and repository contents.
* Add new ``github.create_file`` action which allows user to create new files in repositories.
* Add new ``github.update_file`` action which allows user to update existing files in repositories.
* Allow passing base64 encoded content to update and create file actions.
* Bump libs version in requirements.txt

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
 # Changelog
 
+## 1.2.2
+
+* Allow passing base64 encoded content to update and create file actions.
```
Comment (nmaludy) [inline CHANGES.md]: why not just combine these into 1.2.0?

File: README.md [hunk 2]
```diff
@@ -107,6 +107,9 @@ StackStorm webhook handler.
 * ``list_pulls`` - List all pull requests for a particular repo.
 * ``review_pull`` - Create a review for the provided pull request.
 * ``merge_pull`` - Merge the provided pull request.
```
Comment (nmaludy) [inline README.md:110]: probably good to list the actions in alphabetical order?

File: actions/lib/formatters.py [hunk 3]
```diff
@@ -116,36 +120,93 @@ def pull_to_dict(pull):
...
+
+
+def decode_base64(data):
+    """Decode base64, padding being optional.
+
+    :param data: Base64 data as an ASCII byte string
+    :returns: The decoded byte string.
+
+    """
+    missing_padding = len(data) % 4
```
Comment (nmaludy) [inline actions/lib/formatters.py:200]: this seems a bit wonky, a comment here would be nice

File: actions/lib/utils.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+from github.GithubObject import NotSet
+
+__all__ = [
+    'prep_github_params_for_file_ops'
+]
+
+
+def create_github_author(s):
+    # expecting string in the format of "FirstName LastName <email@address>"
+    pattern = re.compile(r"^(.*?)\s+<([a-zA-Z0-9_.+-]+?@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+)?>$")
```
Comment (nmaludy) [inline actions/lib/utils.py]: for performance reasons, probably good to compile this at the file level and the reuse the compiled pattern within this function

File: actions/lib/utils.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        email = match.group(2)
+        return InputGitAuthor(name=name, email=email)
+    else:
+        return None
+
+
+def prep_github_params_for_file_ops(author, branch, committer):
+    if not branch:
+        branch = NotSet
+    if committer:
```
Comment (nmaludy) [inline actions/lib/utils.py:27]: any reason you don't do if/else here?

**Review by nmaludy — CHANGES_REQUESTED** [review]
Couple comments inline with the code.

General comments:
- Would be nice to see unit tests here
- Let's just use version 1.2.0 for this instead of 1.2.2 since all of the changes are in one PR

Thanks for the contribution!

File: sensors/github_repository_sensor.yaml [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
-          actor:
-            type: "object"
-          payload:
-            type: "object"
+class_name: "GithubRepositorySensor"
+entry_point: "github_repository_sensor.py"
+description: "Sensor which monitors Github repository for activity"
+# requires authentication because of GitHub rate limiting
+# use default with auth or increase poll interval
+enabled: false
```
Comment (arm4b) [inline sensors/github_repository_sensor.yaml]: This sensor was enabled by default. Let's please keep it enabled.
```suggestion
enabled: true
```

File: CHANGES.md (hunk 1 above)
Comment (igcherkaev) [reply CHANGES.md]: Was deploying/testing in my dev, got a habit of bumping version each time. Sorry! Didn't think that bumping up build/bugfix version makes any difference to anyone.

File: README.md (hunk 2 above)
Comment (igcherkaev) [reply README.md:110]: :)

point taken

File: actions/lib/utils.py (hunk 4 above)
Comment (igcherkaev) [reply actions/lib/utils.py]: OK.

File: actions/lib/utils.py (hunk 5 above)
Comment (igcherkaev) [reply actions/lib/utils.py:27]: no reason, probably was adding that later and did not pay attention.

File: actions/lib/formatters.py (hunk 3 above)
Comment (igcherkaev) [reply actions/lib/formatters.py:200]: I dunno if it's still the case, but it stems from times where decoding with incorrect padding would give you an exception, see https://gist.github.com/perrygeo/ee7c65bb1541ff6ac770 or https://stackoverflow.com/questions/2941995/python-ignore-incorrect-padding-error-when-base64-decoding

Just to be safe.

File: CHANGES.md (hunk 1 above)
Comment (igcherkaev) [reply CHANGES.md]: resetting it to 1.2.0

**Comment by igcherkaev** [comment]
> Couple comments inline with the code.
> 
> General comments:
> 
> * Would be nice to see unit tests here
> * Let's just use version 1.2.0 for this instead of 1.2.2 since all of the changes are in one PR
> 
> Thanks for the contribution!

* I honestly have very little clue on unit testing and stuff, since I am not a developer per se :( I'd leave it to python gurus.
* Let's use 1.2.0, I have no problems with it.

Thanks for reviewing the PR.

**Review by nmaludy — APPROVED** [review]
LGTM thank you!

**Pull request merged**